src/MainUpdate.elm

```diff
@@ -510,3 +511,52 @@ update msg model =
 
             GetDateAndThenAddDate id ->
                 model ! [ Task.perform (AddNewDate id) Date.now ]
+
+            DeleteTrack trackId ->
+                let
+                    newTracks =
+                        List.filter (\t -> t.id /= trackId) model.tracks
+
+                    apiUpdate =
+                        { datesWithSessions = model.datesWithSessions
+                        , tracks = newTracks
+                        , columns = model.columns
+                        }
+                in
+                    ( { model | tracks = newTracks }
+                    , Api.postModelToDb apiUpdate model.eventId
+                    )
+
+            AddNewTrack ->
+                ( { model
+                    | pickedTracks =
+                        appendNewElementToList model.pickedTracks (Track 0 "" "")
+                  }
+                , Cmd.none
+                )
+
+            UpdatePickedTrack trackId trackInputType newPickedTrackInput ->
+                let
+                    pickedTrack =
+                        model.pickedTracks
+                            |> List.filter (\t -> t.id == trackId)
+                            |> List.head
+                            |> Maybe.withDefault (Track 0 "" "")
+
+                    newPickedTrack =
+                        if trackInputType == "name" then
```

Comment:
They're easier to follow for someone else reading the code, allow elm compiler to check our logic and prevent typo bugs